robot_params:
     - joint: "shoulder_pan"
       type: "revolute"
       position_upper: 3.14159
       position_lower: -3.14159
       velocity: 1.57
       angular_acceleration: 1.1
       envelope_positions:
       - [0.0,0.0,0.0]
       envelope_radius:
       - 80.0

     -  joint: "shoulder_lift"
        type: "revolute"
        position_upper: 0.0
        position_lower: -3.14159
        velocity: 1.57
        angular_acceleration: 1.5
        envelope_positions:
        - [0.0,0.0,0.0]
        - [110.0,0.0,0.0]
        - [210.0,0.0,0.0]
        - [310.0,0.0,0.0]
        - [420.0,0.0,0.0]
        envelope_radius:
        - 76.0
        - 60.0
        - 56.0
        - 56.0
        - 74.0

     -  joint: "elbow"
        type: "revolute"
        position_upper: 1.57
        position_lower: -1.57
        velocity: 1.57
        angular_acceleration: 1.8
        envelope_positions:
        - [10.0,0.0,-115.0]
        - [110.0,0.0,-134.0]
        - [210.0,0.0,-134.0]
        - [309.0,0.0,-134.0]
        - [385.0,0.0,-134.0]
        envelope_radius:
        - 70.0
        - 55.0
        - 55.0
        - 55.0
        - 65.0

     -  joint: "wrist_1_link"
        type: "revolute"
        position_upper: 3.14159
        position_lower: -3.14159
        velocity: 1.57
        angular_acceleration: 2.7
        envelope_positions:
        - [0.0,6.0,-10.0]
        envelope_radius:
        - 55.0

     -  joint: "wrist_2_link"
        type: "revolute"
        position_upper: 3.14159
        position_lower: -3.14159
        velocity: 1.57
        angular_acceleration: 2.7
        envelope_positions:
        - [0.0,-4.0,-10.0]
        envelope_radius:
        - 60.0

     -  joint: "wrist_3_link"
        type: "revolute"
        position_upper: 3.14159
        position_lower: -3.14159
        velocity: 1.8
        angular_acceleration: 2.0
        envelope_positions:
        - [0.0,0.0,0.0]
        envelope_radius:
        - 40.0
